include/transaction/transaction.h

```diff
@@ -6,24 +6,24 @@
 #include "account/address.h"
 #include "transaction/signer.h"
 
-#include <stdint.h>
+#include <cstdint>
 #include <string>
 
 class Transaction
 {
 public:
     Transaction(
-            uint64_t nonce,
-            std::string value,
-            Address receiver,
-            Address sender,
-            uint64_t gasPrice,
-            uint64_t gasLimit,
-            std::string data,
-            std::string chainID,
-            uint64_t version);
-
-    void applySignature(Signer signer);
+            uint64_t const &nonce,
+            std::string const &value,
+            Address const &receiver,
+            Address const &sender,
+            uint64_t const &gasPrice,
+            uint64_t const &gasLimit,
```

Comment:
A bit off-topic - a SDK user might often want to build a large set of transactions (say, hundreds or even thousands) without actually knowing their nonces - that is, the SDK user might want to build the transactions, but delay setting the `nonce` and applying the signatures. I think the same would go for `gasPrice` as well. Perhaps we should handle this scenario in a future PR (currently, the user can only provide these transaction fields within the constructor, and cannot operate on / with partially constructed transactions)?

Reply:
Very nice explanation, appreciate the details.
The reason I've made them `const &` is because **CLion IDE Clang-Tidy** complained about the parameters, for instance see this case where I've only deleted const & for receiver's Address:
![image](https://user-images.githubusercontent.com/82832880/121011410-3f42c500-c79f-11eb-9be1-4129b1233cdd.png)

I'm also not sure which approach we should take, however your opinion seems solid. Do you think we should pass them by value instead?

Reply:
Regarding constructing transactions without nonce and gasPrice: you are right, good point, we should take care of it in the future. This topic is already on the table from previous discussions with @AdoAdoAdo, I will also add a TODO here to keep track of it.